### ISSUE TITLE
Store cookies and local storage data to avoid repetitive login

### ISSUE THREAD

**Issue by li-xinyang** [linkedtales/scrapedin#29]
To avoid repetitive login process, the script could store cookies and local storage data in temporary files after the first time login and use them in the subsequent logins if the credentials are still valid so that the users of the module can avoid repetitive login and reduce the chances get their account blocked.

If the credentials in the temporary files have expired the page would be redirected to the default login page the script can repeat the process again.

This also accelerates the crawling process because the login setup can be skipped.

**Comment by imnamanmodi1** [comment]
I totally agree, I am building a real scraper out of it & it is just crushing my logins, why do we have to do repitative logins

**Comment by leonardiwagner** [comment]
It's a good idea indeed, just send a PR and I'm going to merge it.

**Comment by imnamanmodi1** [comment]
@leonardiwagner will send a PR, I was just confirming will it be right if I code in login.js, I think that's the place to add it.

**Comment by leonardiwagner** [comment]
@imnamanmodi1 No, because `login.js` is only about the login page only.

This is probably related to `scrapedin.js` or  `openPage.js` , to attach the cookie to the browser/page instace.

And if cookies is provided we acutally should to bypass login [here](https://github.com/linkedtales/scrapedin/blob/master/src/scrapedin.js#L22)

**Comment by fraywing** [comment]
@leonardiwagner @imnamanmodi1 is there a way we can just use this to scrape single public profiles without login or check? It would be useful if in addition to batch crawl and login scrape to have a non-login targeted public scrape function. 

That way we can use this headless on a remote server

**Comment by imnamanmodi1** [comment]
@fraywing
I don't think so because when you don't login, you won't be able to see full profiles, Linkedin has this restriction thus login is necessary.
Btw I have built a scraper which runs in loop & can scrap data by reading a .csv file.

I have also added two simultaneous login credentials to be used while scraping in batches so that LinkedIn do not block us considering our login attempts & browsing activity.

If you would like to take a look, I will upload my code on GitHub & share the link here :)

My scraper reads CSV file, scraps the data of each LinkedIn profile url in that CSV & convert it to a JSON file.

Let me know if I can be of any help to you.

& I think you can still run this headless on a remote server without need of opening a browser each time you scrape
there is an option in script where you can define, isHeadless: true & that's it :)

**Comment by fraywing** [comment]
Thanks for replying so quickly! I'd love to see the CSV batch code. 

> & I think you can still run this headless on a remote server without need of opening a browser

Yes, but I couldn't just deploy this to a remote server because of the manual check, right?

I could copy the cookie somehow and run it on the IP address, but the cookie would expire eventually and it would break on remote until we somehow got a new cookie.

It makes it difficult to have an autoscraping API, as you can truly only run this locally. (Unless you came up with some automated way to renew the cookie, or use something like this: https://www.npmjs.com/package/puppeteer-extra-plugin-recaptcha to automatically bypass the captcha?)

**Comment by leonardiwagner** [comment]
Hi guys, I just released `v1.0.9` and now scrapedin works with cookies, please update your version and read the tutorial: https://github.com/linkedtales/scrapedin/wiki/Using-Cookies-To-Login

Please give some feedback, and feel free to reopen this issue if it isn't really solved.
